src/AstType.h

```diff
@@ -63,22 +63,16 @@ class AstPrimitiveType : public AstType {
     /** Creates a new primitive type */
     AstPrimitiveType(const AstQualifiedName& name, TypeAttribute type) : AstType(name), type(type) {}
 
-    /** Tests whether this type is a numeric type */
-    bool isNumeric() const {
-        return isNumericType(type);
-    }
-
-    /** Tests whether this type is a symbolic type */
-    bool isSymbolic() const {
-        return type == TypeAttribute::Symbol;
-    }
-
     AstPrimitiveType* clone() const override {
         auto res = new AstPrimitiveType(getQualifiedName(), type);
         res->setSrcLoc(getSrcLoc());
         return res;
     }
 
+    TypeAttribute getTypeAttribute() const {
+        return type;
+    }
+
 protected:
     void print(std::ostream& os) const override {
         os << ".type " << getQualifiedName() << (type == TypeAttribute::Signed ? "= number" : "");
```

Comment:
There might be the question of whether we should allow subtyping only for primitive types or for any type.

Reply:
What would be the behaviour for records? Right now records behave nominally.

Aside: There's also the related question of `.type A`, which is currently an alias for `.symbol_type A`. Would it be preferable for it to declare a nominal opaque type (i.e. only definable by EDB) rather than a symbol subset?

Reply:
For the time being, I would avoid constructing sub-types of records or sub-types of sub-types -- the type-system could cope with these extensions but let's focus on the necessities, i.e., completing union types for primitive types such as signed, unsigned, float and symbols (at the moment it would only work partially for number/symbol types). 

Indeed -- ```.type A``` should be deprecated as well. 

I would suggest getting the basic type-system into a more solid state and extend it later with opaque types and equivalence types if the need arises.

We may want to create new issues for these two different extensions of the type system.